echo/test/unit/echo_client_test.js

```diff
@@ -90,88 +90,123 @@ describe("Echo Node Client Test Suite", function(){
                 echo_endpoint: endPoint
             });
 
-            var requestStub = sandbox.stub(request, 'post');
-            requestStub.yields(new Error('Error communicating with Echo'));
+            var mockHttpRequest = nock(endPoint)
+                .post('/1/events')
+                .reply(400, {});
 
             echoClient.addEvents('secret', {class:'class', source:'source'}, function(err, result){
                 (err === null).should.be.false;
-                err.code.should.equal(1);
-                err.label.should.equal('REQUEST_ERROR');
+                err.code.should.equal(2);
+                err.label.should.equal('INVALID_QUERY');
```

Comment:
The mocking of the response has caused this change to the error code from "REQUEST_ERROR" to "INVALID_QUERY".